src/features/dsComponents/body/BodyTable.tsx

```diff
@@ -1,226 +1,219 @@
-  { width: 163.77, component: <rect y='13' x='8.62' width="113" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 113.7, component: <rect y='13' x='8.62' width="61" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 98.03, component: <rect y='13' x='8.62' width="34" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 124, component: <rect y='13' x='8.62' width="42" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 109.19, component: <rect y='13' x='8.62' width="57" height="14" rx="1" fill="#D5DADD"/> },
-  { width: 75.13, component: <rect y='13' x='8.62' width="43" height="14" rx="1" fill="#D5DADD"/> }
+  { width: 185.2, component: <rect y='1' x='8.62' width="85" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 163.77, component: <rect y='1' x='8.62' width="113" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 113.7, component: <rect y='1' x='8.62' width="61" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 98.03, component: <rect y='1' x='8.62' width="34" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 124, component: <rect y='1' x='8.62' width="42" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 109.19, component: <rect y='1' x='8.62' width="57" height="15" rx="1" fill="#D5DADD"/> },
+  { width: 75.13, component: <rect y='1' x='8.62' width="43" height="15" rx="1" fill="#D5DADD"/> }
 ]
 
-export default class BodyTable extends React.Component<BodyTableProps> {
-  constructor (props: BodyTableProps) {
-    super(props)
-    this.state = {
-      bodyLength: 0
-    }
+const BodyTable: React.FC<BodyTableProps> = ({
+  rowCount,
+  headers = [],
+  qriRef,
+  loading = false
+}) => {
+  const [ list, setList ] = useState<Record<number, any>>({})
+  const [tableContainer, {
+    height: tableContainerHeight,
+    width: tableContainerWidth
+  }] = useDimensions()
+
+  const timeoutRef: { current: number } = useRef(0)
+
+  const isRowLoaded = ({ index }: any) => {
+    return !!list[index]
+  }
 
-    // // use underscore to debounce the onScroll event
-    // this.handleVerticalScrollThrottled = _.debounce(this.handleVerticalScroll, 250)
+  const loadMoreRows = async ({ startIndex, stopIndex }: any) => {
+    clearTimeout(timeoutRef.current)
+    return await new Promise<void>((resolve, reject) => {
+      // debounce this function so user can scroll way down the table and not trigger hundreds of API calls
+      timeoutRef.current = window.setTimeout(() => {
+        fetch(`https://rosebud-api.qri.cloud/ds/get/${qriRef.username}/${qriRef.name}/body?limit=100&offset=${parseInt(stopIndex) - PAGE_SIZE + 1}`)
+          .then(async (d) => await d.json())
+          .then((d) => {
+            const newList = {}
+            d.data.forEach((row: Array<Record<string, any>>, i: number) => {
+              list[parseInt(startIndex) + i] = row
+            })
+
+            setList({
+              ...list,
+              ...newList
+            })
+
+            resolve()
+          })
```

Comment:
If we refactor `loadBody`'s signature to use limit and offset instead of page and pageSize:

`export function loadBody(ref: QriRef, offset: number = 0, Limit: number = DEFAULT_BODY_PAGE_SIZE): ApiActionThunk`

& refactor `fetchBody` to also use offset and limit:

```
function fetchBody (ref: QriRef, offset: number, limit: number): ApiAction {
  return {
    type: 'body',
    ref,
    [CALL_API]: {
      endpoint: 'ds/get',
      method: 'GET',
      segments: {
        username: ref.username,
        name: ref.name,
        path: ref.path,
        selector: ['body'],
        limit,
        offset
      }
    }
  }
}
```

We _should_ be able to replace the fetch above with 

`loadBody(qriRef, parseInt(stopIndex) - DEFAULT_BODY_PAGE_SIZE + 1)`